src/app/shared/document/document-detail/document-detail.component.ts

```diff
@@ -17,379 +17,128 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import {Component, EventEmitter, Input, OnDestroy, OnInit, Output} from '@angular/core';
+import {Component, Input, OnChanges, OnDestroy, OnInit, SimpleChanges} from '@angular/core';
 import {I18n} from '@ngx-translate/i18n-polyfill';
 import {NotificationService} from '../../../core/notifications/notification.service';
-import {AttributeModel, CollectionModel} from '../../../core/store/collections/collection.model';
+import {CollectionModel} from '../../../core/store/collections/collection.model';
 import {DocumentModel} from '../../../core/store/documents/document.model';
-import {Subscription, Observable, interval} from 'rxjs';
+import {Observable, Subscription} from 'rxjs';
 import {Store} from '@ngrx/store';
 import {AppState} from '../../../core/store/app.state';
-import {selectDocumentById} from '../../../core/store/documents/documents.state';
-import {DocumentsAction} from '../../../core/store/documents/documents.action';
-import {selectCollectionById} from '../../../core/store/collections/collections.state';
 import {selectUserById} from '../../../core/store/users/users.state';
 import {filter, map, take} from 'rxjs/operators';
 import {isNullOrUndefined} from 'util';
 import {UsersAction} from '../../../core/store/users/users.action';
 import {selectOrganizationByWorkspace} from '../../../core/store/organizations/organizations.state';
-import {CorrelationIdGenerator} from '../../../core/store/correlation-id.generator';
-import {CollectionsAction} from '../../../core/store/collections/collections.action';
-import {DetailRow} from '../detail-row';
+import {DocumentUiService} from '../../../core/ui/document-ui.service';
+import {DocumentsAction} from '../../../core/store/documents/documents.action';
+import {UiRow} from '../../../core/ui/ui-row';
+import DeleteConfirm = DocumentsAction.DeleteConfirm;
 
 @Component({
   selector: 'document-detail',
   templateUrl: './document-detail.component.html',
-  styleUrls: ['./document-detail.component.scss']
+  styleUrls: ['./document-detail.component.scss'],
+  // changeDetection: ChangeDetectionStrategy.OnPush
```

Comment:
Why is this commented? Does not it work with OnPush change detection?

Reply:
Still need to figure out. Will be investigated in the subsequent PR.